cql3/expr/prepare_expr.cc

```diff
@@ -1000,9 +1000,39 @@ try_prepare_expression(const expression& expr, data_dictionary::database db, con
                 if (!prepared_arg_opt) {
                     return std::nullopt;
                 }
+
```

Comment:
A note about the commit message: I think it's not true that you need to pass the "partition key columns" to token(). You usually do, but don't have to. Unless I'm misunderstanding something, what you really need to do is to pass values matching those columns' types.

Reply:
You're right: the following query works:

```cql
select * from system.local where token(key) = token('local');
```

Reply:
Huh I guess any `RHS` `token()` gets replaced with an anonymous function somewhere along the way:
```
preparing binary operator: (token(unresolved(p))) = token(?)
prepared: (token(p)) = <anonymous function>(?)
```
I'm not sure if it's something that we really want, will investigate.

Reply:
In general not, anonymous functions destroy information.

But it's of minor importance, nothing bad happens if we keep using a function.